onnx/defs/tensor/defs.cc

```diff
@@ -2461,6 +2461,302 @@ ONNX_OPERATOR_SET_SCHEMA(
+                  start_h, step_h, start_w, step_w = If (constant_align_corners_equal_zero) <
+                      then_branch = h1 () => (start_h_then, step_h_then, start_w_then, step_w_then) { # => (float, float, float, float)
+                          step_h_then = Div (two, H_float)
+                          step_w_then = Div (two, W_float)
+                          step_h_half = Div (step_h_then, two)
+                          start_h_then = Add (minus_one, step_h_half)
+                          step_w_half = Div (step_w_then, two)
+                          start_w_then = Add (minus_one, step_w_half)
+                      },
+                      else_branch = h2 () => (start_h_else, step_h_else, start_w_else, step_w_else) { # => (float, float, float, float)
+                          H_float_nimus_one = Sub (H_float, one)
+                          W_float_nimus_one = Sub (W_float, one)
+                          step_h_else = Div (two, H_float_nimus_one)
+                          step_w_else = Div (two, W_float_nimus_one)
+                          start_h_else = Identity (minus_one)
+                          start_w_else = Identity (minus_one)
+                      }
+                  >
+                  grid_w_steps_int = Range (int_zero, W, int_one)
+                  grid_w_steps_float = CastLike (grid_w_steps_int, step_w)
+                  grid_w_steps = Mul (grid_w_steps_float, step_w)
+                  grid_w_0 = Add (start_w, grid_w_steps)
+
+                  grid_h_steps_int = Range (int_zero, H, int_one)
+                  grid_h_steps_float = CastLike (grid_h_steps_int, step_h)
+                  grid_h_steps = Mul (grid_h_steps_float, step_h)
+                  grid_h_0 = Add (start_h, grid_h_steps)
+
+                  zeros_W_by_H = Transpose (zeros_H_by_W)
+                  grid_h_1 = Add (zeros_W_by_H, grid_h_0)
+                  grid_h = Transpose (grid_h_1)
+
+                  grid_w = Add (grid_w_0, zeros_H_by_W)
+
+                  # make following a function (theta, grid_w, grid_h) =>  (grid)
+                  original_grid_seq = SequenceConstruct (grid_w, grid_h, ones_H_by_W)
+                  original_grid = ConcatFromSequence <axis: int=-1, new_axis: int=1> (original_grid_seq)
+                  constant_shape_HW_3 = Constant <value_ints: ints = [-1, 3]> ()
+                  original_grid_HW_3 = Reshape (original_grid, constant_shape_HW_3)
+                  original_grid_3_HW_ = Transpose (original_grid_HW_3)
+
+                  original_grid_3_HW = CastLike (original_grid_3_HW_, theta)
+                  grid_N_2_HW = MatMul (theta, original_grid_3_HW)
+                  grid_N_HW_2 = Transpose <perm = [0, 2, 1]> (grid_N_2_HW)
+                  N_H_W_2_seq = SequenceConstruct (N, H, W, int_two_1d)
+                  N_H_W_2 = ConcatFromSequence <axis: int=-1, new_axis: int=0> (N_H_W_2_seq)
+                  grid_2d_then_ = Reshape(grid_N_HW_2, N_H_W_2)
```

Comment:
Consistency: white space after op? Here and elsewhere

```suggestion
                  grid_2d_then_ = Reshape (grid_N_HW_2, N_H_W_2)
```

Reply:
I noticed this convention before. I wonder reason behind this and do we want to enforce it? To me without space is more nature. @gramalingam ?

Reply:
I agree consistency is good. I personally prefer the space, especially for ONNX here (no nested expressions, only a single call, so it works well).

Reply:
Sure, I updated with the space convention. To summarize it: open brackets shall be preceded with a single space.